pkg/image/definition.go

```diff
@@ -69,6 +69,7 @@ type OperatingSystem struct {
 }
 
 type Packages struct {
+	NoGPGCheck      bool      `yaml:"noGPGCheck"`
```

Comment:
We decided to move away from the more granular `gpg-check` approach, mainly for two reasons:

1. It will offer significant complexity to the code for something that we do not want to encourage
2. If we do it this way we essentially encourage the usage of non-signed repos/rpms along with signed one's. Which IMO is a disaster waiting to happen

Disabling GPG checks should be done for dev/test purposes only and by having only a global switch IMO this will enforce users to GPG sign their repos/rpms more than having a granular approach to GPG validation.

This will be thoroughly documented in the EIB RPM resolver documentation PR which I am currently working on.

Reply:
Thanks for the clarification! Do we want to then fail the build by default if `gpg-keys` dir is not present in order to "enforce" this default behaviour?

Reply:
So currently:

1.  If the RPM dir is present and the `noGPGCheck: true`, then we do not fail and we do all installation options with `--no-gpg-check`
2. If the RPM dir is present, but no GPG dir is present and `noGPGCheck: false`, then the build will fail because the validation of the RPMs will not succeed - RPMs will either be missing signatures, or have an unrecognised (non-imported) key signature.

There is also a corner case in that you may side-load an RPM whose GPG key comes from a third-party repository that you have added to the EIB configuration, then the build will not fail, because the GPG key will be already present on the machine since it will be added from the repo itself (using the `--gpg-auto-import-keys`).

So that said I think that failing the build by default will brake this corner case. I think that the current validation is enough, WDYT?

Reply:
That said I can definitely validate and fail if all of the below is true:

1. `additionalRepositories` is missing
2. `noGPGCheck: false`
3. `gpg-keys` directory is missing

Adding this to my TODO.